tests/test_pages.py

```diff
@@ -60,3 +72,100 @@ def to_item(self) -> dict:
     assert is_injectable(MyItemPage()) is False
     assert is_injectable(ItemPage) is True
     assert is_injectable(ItemWebPage) is True
+
+
+@pytest.mark.asyncio
+async def test_item_page_typed():
+    class MyPage(ItemPage[Item]):
+        @field
+        def name(self):
+            return "name"
+
+    page = MyPage()
+    assert page.item_cls is Item
+    item = await page.to_item()
+    assert isinstance(item, Item)
+    assert item == Item(name="name")
+
+
+@pytest.mark.asyncio
+async def test_web_page_fields():
+    class MyPage(WebPage[Item]):
+        @field
+        def name(self):
+            return "name"
+
+    page = MyPage(HttpResponse(url="http://example.com", body=b""))
+    assert page.item_cls is Item
+    item = await page.to_item()
+    assert isinstance(item, Item)
+    assert item == Item(name="name")
+
+
+@pytest.mark.asyncio
+async def test_item_page_typed_subclass():
+    class BasePage(ItemPage[ItemT]):
+        @field
+        def name(self):
+            return "name"
+
+    class Subclass(BasePage[Item]):
+        pass
+
+    page = BasePage()
+    assert page.item_cls is dict
+    assert (await page.to_item()) == {"name": "name"}
+
+    page2 = Subclass()
+    assert page2.item_cls is Item
+    assert (await page2.to_item()) == Item(name="name")
+
+
+@pytest.mark.asyncio
+async def test_item_page_change_item_type_extra_fields() -> None:
+    class BasePage(ItemPage[Item]):
+        @field
+        def name(self):
+            return "hello"
+
+    @attrs.define
+    class MyItem(Item):
+        price: float
+
+    class Subclass(BasePage, Returns[MyItem]):
+        @field
+        def price(self):
+            return 123
+
+    page = Subclass()
+    assert page.item_cls is MyItem
+    item = await page.to_item()
+    assert isinstance(item, MyItem)
+    assert item == MyItem(name="hello", price=123)
+
+
+@pytest.mark.asyncio
+async def test_item_page_change_item_type_remove_fields() -> None:
+    @attrs.define
+    class MyItem:
+        name: str
+        price: float
+
+    class BasePage(ItemPage[MyItem]):
+        @field
+        def name(self):
+            return "hello"
+
+        @field
+        def price(self):
+            return 123
+
+    # Item only contains "name", but not "price"
+    class Subclass(BasePage, Returns[Item], skip_nonitem_fields=True):
```

Comment:
Let's also add a test similar to this but `skip_nonitem_fields` is left out being `False` to assert that an exception is raised because of it.

Reply:
A good call, added more tests in https://github.com/scrapinghub/web-poet/pull/74/commits/98c2627d0b38ccd56c1c241661dee044db1eb0bc